File: scripts/get_java_jars.pl
#!/usr/bin/perl -w

use strict;
use diagnostics;
use File::Basename;

=head ABOUT

This script reads an Eclipse .classpath file, finds all lib entries
in it and outputs the classpath in various forms needed for various
tools.

=cut

=head TODO

- turn this into python.
- use a standard XML parser instead of cat/grep combo.

=cut

if(scalar(@ARGV)!=1) {
	die("usage: prog [base_jars|jnlp|jars|classpath]");
}

my($cmd)=$ARGV[0];
#my($str)=`cat .classpath | grep kind=\"lib\" | cut -d \\\" -f 4`;
my(@array)=`cat .classpath | grep kind=\\\"lib | cut -d \\\" -f 4`;

for(my($i)=0;$i<@array;$i++) {
	chop($array[$i]);
	if($cmd eq "base_jars") {
		my($name,$path,$suffix)=fileparse($array[$i],[".jar"]);
		$array[$i]=$name;
	}
	if($cmd eq "jnlp") {
		my($name,$path,$suffix)=fileparse($array[$i],[".jar"]);
		$array[$i]="<jar href=\\\"".$name."\\\"/>";
	}
	if($cmd eq "classpath") {
		# don't do anything
		#$array[$i]=$array[$i];
	}
}

if($cmd eq "jars" or $cmd eq "base_jars") {
	print join(" ",@array);
}
if($cmd eq "classpath") {
	print join(":",@array);
}
if($cmd eq "jnlp") {
	print join("\n",@array);
}

File: scripts/get_java_jars.py
#!/usr/bin/env python3

"""
This script reads an Eclipse .classpath file, finds all lib entries
in it and outputs the classpath in various forms needed for various
tools.

TODO
- Use a more robust XML parsing method if the .classpath file structure is complex.
"""

import sys
import os
import xml.etree.ElementTree as ET


def main():
    """ main code """
    if len(sys.argv) != 2:
        print("Usage: python script.py [base_jars|jnlp|jars|classpath]")
        sys.exit(1)

    command = sys.argv[1]
    # Parse the .classpath XML file
    tree = ET.parse('.classpath')
    root = tree.getroot()

    # Find all 'classpathentry' elements with kind="lib"
    lib_entries = [entry.get('path') for entry in root.findall(".//classpathentry[@kind='lib']")]

    result = []
    for entry in lib_entries:
        if command == "base_jars":
            result.append(os.path.splitext(os.path.basename(entry))[0])
        elif command == "jnlp":
            result.append(f'<jar href="{os.path.basename(entry)}"/>')
        elif command == "classpath":
            result.append(entry)
        elif command == "jars":
            result.append(entry)

    if command in ["jars", "base_jars"]:
        print(" ".join(result))
    elif command == "classpath":
        print(":".join(result))
    elif command == "jnlp":
        print("\n".join(result))

if __name__ == "__main__":
    main()
